### PR TITLE
Deploy

### PR DESCRIPTION
**Pull request by cmccully**
Added the necessary components to deploy using kustomize and eventually argocd.

### CODE REVIEW

File: local-kubeconfig [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
```
Comment (jashan-lco) [inline local-kubeconfig:1]: This file should be removed & added to `.gitignore`. I'll update the template to make this note.

File: k8s/base/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - ./deploy-server.yaml
+  - ./svc-server.yaml
+  
+configMapGenerator:
```
Comment (jashan-lco) [inline k8s/base/kustomization.yaml:8]: This should go in the overlay in `k8s/envs/local/kustomization.yaml`

File: k8s/base/settings.env [hunk 3]
```diff
@@ -0,0 +1,2 @@
+ARCHIVE_URL=https://archive-api.lco.global/frames/
```
Comment (jashan-lco) [inline k8s/base/settings.env:1]: This file should be moved to `k8s/envs/local/settings.env`

**Review by jashan-lco — CHANGES_REQUESTED** [review]

File: k8s/base/settings.env (hunk 3 above)
Comment (cmccully) [reply k8s/base/settings.env:1]: Does this need to be in local settings if this is the default value for all deploys and you only have to override it in staging?

File: k8s/base/kustomization.yaml (hunk 2 above)
Comment (cmccully) [reply k8s/base/kustomization.yaml:8]: Same as below?

File: k8s/base/settings.env (hunk 3 above)
Comment (jashan-lco) [reply k8s/base/settings.env:1]: If you'd like these to be the default, then you have to inline them using the `literal` form:

```yaml
configMapGenerator:
  - name: env
    literals:
      - ARCHIVE_URL=https://archive-api.lco.global/frames/
      - ETC=xxx
```

Reason being, any non-yaml files are not copied around by `kpt pkg get ...`

Comment (cmccully) [reply k8s/base/settings.env:1]: Ok. Will do.

**Pull request closed without merging**